src/app/pages/client/ClientNonUIFeatures.tsx

```diff
@@ -204,7 +204,14 @@ function MessageNotifications() {
       const cachedUnreadInfo = unreadCacheRef.current.get(room.roomId);
       unreadCacheRef.current.set(room.roomId, unreadInfo);
 
-      if (unreadInfo.total === 0) return;
+      if (unreadInfo.total === 0) {
+        const handleUnreadEvent: RoomEventHandlerMap[RoomEvent.UnreadNotifications] = () => {
+          handleTimelineEvent(mEvent, room, toStartOfTimeline, removed, data);
+          room.removeListener(RoomEvent.UnreadNotifications, handleUnreadEvent);
+        };
+        room.on(RoomEvent.UnreadNotifications, handleUnreadEvent);
```

Comment:
since you are removing the listener inside event handler, it feels like to me that following change may lead to memory leaks where the event handler is set but never gets removed?

Reply:
Yes, that is something I did not consider. I don't really know how to tackle this - my only idea would be to remove the listener after some time. Or I could change it such that for each room there will be at most one listener which would still add memory overhead but constant in the number of rooms the user joined :thinking: